src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas2/ApplicationNoteService.kt

```diff
@@ -17,17 +22,22 @@ class ApplicationNoteService(
   private val applicationRepository: Cas2ApplicationRepository,
   private val applicationNoteRepository: Cas2ApplicationNoteRepository,
   private val userService: NomisUserService,
+  private val externalUserService: ExternalUserService,
+  private val httpAuthService: HttpAuthService,
 ) {
 
   fun createApplicationNote(applicationId: UUID, note: NewCas2ApplicationNote):
     AuthorisableActionResult<ValidatableActionResult<Cas2ApplicationNoteEntity>> {
     val application = applicationRepository.findByIdOrNull(applicationId)
       ?: return AuthorisableActionResult.NotFound()
 
-    val user = userService.getUserForRequest()
+    val isExternalUser = httpAuthService.getCas2AuthenticatedPrincipalOrThrow().isExternalUser()
+    val user = getExternalOrNomisUser(isExternalUser)
```

Comment:
Just from reading this in order I wouldn't be sure what type of thing  `isExternalUser` becomes? Could be boolean from the 'is' prefix but then it's passed as a param which suggests it could be object?

Reply:
Would it make sense for the getExternalOrNomisUser method to also take responsibility for this logic too? Seems related to finding a user. I'm guessing that logic is here so that isExternalUser is available later on for the conditional?

```
val user = getExternalOrNomisUser
if (!user.isExternal){
…
}

  private fun getExternalOrNomisUser: Cas2User {
    val isExternalUser = httpAuthService.getCas2AuthenticatedPrincipalOrThrow().isExternalUser()
    
    return if (isExternalUser) {
      {
        externalUserService.getUserForRequest(),
        isExternal: true
      }
    } else {
      {
        userService.getUserForRequest(),
        isExternal: false
      }
    }
  }
 ```
 
I'm not sure exactly how we'd do it but it feels like a useful attribute of a Cas2User object to know internal/external more generally?

Reply:
I agree I think there's some tidying up/shifting around to do, but I'm not sure how exactly to do it either.

>  Could be boolean from the 'is' prefix but then it's passed as a param which suggests it could be object?
Is there something against passing a boolean as a param? not heard that before.

>  it feels like a useful attribute of a Cas2User object to know internal/external more generally?

I think the `isExternalUser` is quite a specific thing- it's based on the `auth_source` of the `jwt` so I think it makes sense for the `httpAuthService` to be the one that decides that (and we also then get to check that it's a valid auth source)... I don't know that the `Cas2User` should know about it, I think if it were a function or field on the `Cas2User` we'd also have to add it the classes that extend, which should definitely not know imo  🤔 

```
...
    return if (isExternalUser) {
      {
        externalUserService.getUserForRequest(),
        isExternal: true
      }
    } else {
      {
        userService.getUserForRequest(),
        isExternal: false
      }
    }
  }
...
```

This would be a good solution but I can't figure out how to do it in a Kotlin way that doesn't make it less readable 😢 

I've pushed a commit with a refactor to the file, I think it makes it more readable.